### PR TITLE
Add known-standard-library for each Python version

### PR DESCRIPTION
**Pull request by JonathanPlasse**
- Add a script to generate `known-standard-library` for each Python version as described in #2343.

### CODE REVIEW

File: src/python/sys.rs [hunk 1]
```diff
@@ -1,228 +1,1108 @@
 use once_cell::sync::Lazy;
-use rustc_hash::FxHashSet;
+use rustc_hash::{FxHashMap, FxHashSet};
+
+use crate::settings::types::PythonVersion;
 
 // See: https://pycqa.github.io/isort/docs/configuration/options.html#known-standard-library
```
Comment (charliermarsh) [inline src/python/sys.rs:6]: Can we add a note that this file is autogenerated?

File: scripts/generate_known_standard_library.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+"""Vendored from [scripts/mkstdlibs.py in PyCQA/isort](https://github.com/PyCQA/isort/blob/e321a670d0fefdea0e04ed9d8d696434cf49bdec/scripts/mkstdlibs.py).
+
+Only the generation of the file has been modified for use in this project.
+"""
+
+from pathlib import Path
+
+from sphinx.ext.intersphinx import fetch_inventory
```
Comment (charliermarsh) [inline scripts/generate_known_standard_library.py:8]: Do we need to add `sphinx` as a dep?

Comment (JonathanPlasse) [reply scripts/generate_known_standard_library.py:8]: We would have to do the parsing ourselves.
I could do it if you think it is worth it.

Comment (JonathanPlasse) [reply scripts/generate_known_standard_library.py:8]: Isort does not have sphinx as a dependency.
I can add it anyway.

Comment (charliermarsh) [reply scripts/generate_known_standard_library.py:8]: But, did you have to install Sphinx for this to work? I'm confused.

Comment (charliermarsh) [reply scripts/generate_known_standard_library.py:8]: It's alright, I can always fix it later if needed.

Comment (JonathanPlasse) [reply scripts/generate_known_standard_library.py:8]: Yes, I guess they install it manually to run every new Python release.

Comment (JonathanPlasse) [reply scripts/generate_known_standard_library.py:8]: I added it.

**Comment by charliermarsh** [comment]
Great stuff, thank you.

**Comment by JonathanPlasse** [comment]
You are welcome! 😁

**Pull request merged**